ros2action/test/fixtures/fibonacci_action_introspection.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2025 Sony Group Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import rclpy
+from rclpy.action import ActionClient
+from rclpy.action import ActionServer
+from rclpy.executors import ExternalShutdownException, SingleThreadedExecutor
+from rclpy.node import Node
+from rclpy.qos import qos_profile_system_default
+from rclpy.service_introspection import ServiceIntrospectionState
+
+from test_msgs.action import Fibonacci
+
+
+class FibonacciActionServer(Node):
+
+    def __init__(self):
+        super().__init__('fibonacci_action_server', namespace='/test')
+        self._action_server = ActionServer(
+            self,
+            Fibonacci,
+            'fibonacci',
+            self.execute_callback)
+        self._action_server.configure_introspection(
+            self.get_clock(), qos_profile_system_default, ServiceIntrospectionState.CONTENTS)
+
+    def destroy_node(self):
+        self._action_server.destroy()
+        super().destroy_node()
+
+    def execute_callback(self, goal_handle):
+        feedback = Fibonacci.Feedback()
+        feedback.sequence = [0, 1]
+
+        for i in range(1, goal_handle.request.order):
+            feedback.sequence.append(feedback.sequence[i] + feedback.sequence[i-1])
+            goal_handle.publish_feedback(feedback)
+
+        goal_handle.succeed()
+
+        result = Fibonacci.Result()
+        result.sequence = feedback.sequence
+        return result
+
+
+class FibonacciActionClient(Node):
+
+    def __init__(self):
+        super().__init__('fibonacci_action_client', namespace='/test')
+        self._action_client = ActionClient(self, Fibonacci, 'fibonacci')
```

Comment:
why action client does not enable introspection for the unit test? i think we need to check both action server and client service event?

Reply:
Okay. Update code.